.github/workflows/testing.yml

```
@@ -3,16 +3,30 @@ name: Testing
 on:
   push:
     branches: [ master ]
+    paths-ignore:
+      - '**.md'
+      - '**/mkdocs.yml'
+      - '**/shellcheck.yml'
+      - '**/stats.yml'
   pull_request:
     branches: [ master ]
 
 jobs:
   unit:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-20.04, ubuntu-22.04, macos-11, macos-12 ]
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Install os dependencies
-        run: sudo apt update && sudo apt install attr kcov
+        run: |
+          if [ "${{ runner.os }}" = "Linux" ]; then
+            sudo apt update && sudo apt install attr kcov
+          else
+            brew install kcov diffutils grep
```

This might speed up the pipeline execution time a little bit
```suggestion
            HOMEBREW_NO_AUTO_UPDATE=1 brew install kcov diffutils grep
```